bindings/rust/s2n-tls/src/testing/resumption.rs

```diff
@@ -86,59 +86,32 @@ mod tests {
             .set_connection_initializer(handler)?;
         let client_config = client_config_builder.build()?;
 
-        // create and configure a server connection
-        let mut server = connection::Connection::new_server();
-        server
-            .set_config(server_config.clone())
-            .expect("Failed to bind config to server connection");
-
-        // create a client connection
-        let mut client = connection::Connection::new_client();
-
-        // Client needs a waker due to its use of an async callback
-        client
-            .set_waker(Some(&noop_waker()))?
-            .set_config(client_config.clone())
-            .expect("Unable to set client config");
-
-        let server = Harness::new(server);
-        let client = Harness::new(client);
-        let pair = Pair::new(server, client);
-        let pair = poll_tls_pair(pair);
-
-        let client = pair.client.0.connection();
-
-        // Check connection was full handshake and a session ticket was included
-        assert!(!client.resumed());
-        validate_session_ticket(client)?;
-
-        // create and configure a client/server connection again
-        let mut server = connection::Connection::new_server();
-        server
-            .set_config(server_config)
-            .expect("Failed to bind config to server connection");
-
-        // create a client connection with a resumption ticket
-        let mut client = connection::Connection::new_client();
-
-        client
-            .set_waker(Some(&noop_waker()))?
-            .set_config(client_config)
-            .expect("Unable to set client config");
-
-        let server = Harness::new(server);
-        let client = Harness::new(client);
-        let pair = Pair::new(server, client);
-        let pair = poll_tls_pair(pair);
+        // initial handshake, no resumption
+        {
+            let mut pair = TestPair::from_configs(&client_config, &server_config);
+            // Client needs a waker due to its use of an async callback
+            pair.client.set_waker(Some(&noop_waker()))?;
```

Comment:
I'm not really sure why setting a noop waker is needed for an async callback, but is this something the test harness could do?

Reply:
Honestly it's on my list to investigate, because I agree that it seems unnecessary.

While the test harness certainly _could_ do it, I would vote again it. I think we should do the minimal amount of configuration necessary. I'm willing to budge on the blinding configuration because it's so useful, but I worry that automatically setting the no_op waker makes it more difficult to see how annoying this issue it.

Reply:
Agreed, this seems strange. Which of the callbacks on client_config is async? They all look sync to me.

Reply:
I believe the "async" future causing the problem is the connection initializer. The error comes from here
https://github.com/aws/s2n-tls/blob/3eb1bec1fd34c53e2053047c4bc9f9065bfa84b1/bindings/rust/s2n-tls/src/connection.rs#L538-L542

Ideally we would check whether there was actually a future to poll before retrieving the waker, but that isn't the current implementation.

Reply:
That seems like a bug that should be fixed, and you're right that the testing framework was absolutely covering it up :( Can you open an issue?

Reply:
#4608